### DISCUSSION TITLE
Why is the route locations max smaller than optimized_route

### DISCUSSION THREAD

**Discussion by mabilinab** [valhalla/valhalla#4840] [Q&A]
This is a bit random but why is the default 'auto' max_locations for route 20 locations, while the optmized_route can take up to 50 locations? Despite optimized_route also returns a route for all the 50 locations

**Comment by kevinkreiser** [comment]
because the algorithm used to compute the optimized route uses a matrix underneath and the algorithm used to compute regular routes is not. for matrix we set other limits (like distance ones) which are way less generous than regular routes and so you can do more stops but they have to be much much closer together. for regular routes you can cross the US a couple times iirc for matrix you cant really leave the a larger metropolitan area.

**Reply by mabilinab** [reply]
I see! But why isn't it the case for other modes?  e.g bus which has max_locations=50 and the other parameters are the same as auto

**Reply by kevinkreiser** [reply]
good quesiton! perhaps git blame will help us there. i dont have time to spelunk right now but we could do so later. some memory way in the back of my mind is that back when we were developing bus costing we had a specific user that needed more than the other modes and so we just did it for them (remix they were called). also of note is that buses typically (though not greyhound) have shorter multistop routes.
